zcash-hsmbuilder/src/txbuilder.rs

```diff
@@ -137,7 +143,7 @@ where
     pub fn transaction_data(&self) -> Option<TransactionData<A>> {
         self.cached_branchid.map(|consensus_branch_id| {
             TransactionData::from_parts(
-                transaction::TxVersion::Sapling,
+                self.cached_tx_version.expect("No tx version provided"),
```

Comment:
avoid the panic here, rather return `None` if cached_tx is none, just like we would with cached_branchid

Reply:
Should be able to do that with `zip` https://doc.rust-lang.org/std/option/enum.Option.html#method.zip